### PR TITLE
PCE takes into account encapsulation type

### PR DESCRIPTION
**Pull request by timofei-durakov**
This patch adds additional filtering for AvailableNetwork

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2508)

### CODE REVIEW

File: services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/SwitchFeatures.java [hunk 1]
```diff
@@ -74,12 +78,16 @@
     @Property(name = "support_vxlan_vni_match")
     private boolean supportVxlanVniMatch;
 
+    @Property(name = "supported_transit_encapsulation")
+    private Set<FlowEncapsulationType> supportedTransitEncapsulation;
+
     @Builder(toBuilder = true)
     public SwitchFeatures(Switch switchObj, boolean supportBfd, boolean supportVxlanPushPop,
-                          boolean supportVxlanVniMatch) {
+                          boolean supportVxlanVniMatch, Set<FlowEncapsulationType> supportedTransitEncapsulation) {
```
Comment (Settis) [inline services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/SwitchFeatures.java]: It's good idea.

Comment (sergii-iakovenko) [reply services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/SwitchFeatures.java]: Just curious - why not "supportedFlowEncapsulation" ?

Comment (timofei-durakov) [reply services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/SwitchFeatures.java]: wanted to focus on the fact it's about transit rules mostly

File: services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java [hunk 2]
```diff
@@ -174,18 +169,51 @@ public Neo4jIslRepository(Neo4jSessionFactory sessionFactory, TransactionManager
...
+
+        return Lists.newArrayList(getSession().query(getEntityType(), query, parameters));
+    }
+
+    @Override
+    public Collection<Isl> findSymmetricActiveWithAvailableBandwidth(long requiredBandwidth,
+                                                                     FlowEncapsulationType flowEncapsulationType) {
+        String encapsulationType = null;
+        if (flowEncapsulationType != null) {
+            encapsulationType = flowEncapsulationType.toString();
```
[the remark this answers is not in the document]
Comment (timofei-durakov) [reply services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java]: Done

File: services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java [hunk 3]
```diff
@@ -174,18 +169,51 @@ public Neo4jIslRepository(Neo4jSessionFactory sessionFactory, TransactionManager
     }
 
     @Override
-    public Collection<Isl> findSymmetricActiveWithAvailableBandwidth(long requiredBandwidth) {
+    public Collection<Isl> findActiveWithAvailableBandwidth(long requiredBandwidth,
+                                                            FlowEncapsulationType flowEncapsulationType) {
+        String encapsulationType = null;
+        if (flowEncapsulationType != null) {
+            encapsulationType = flowEncapsulationType.toString();
```
[the remark this answers is not in the document]
Comment (timofei-durakov) [reply services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java]: Done

File: services/src/kilda-core/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/SwitchFeaturesRepository.java [hunk 4]
```diff
@@ -15,7 +15,11 @@
 
 package org.openkilda.persistence.repositories;
 
+import org.openkilda.model.Switch;
 import org.openkilda.model.SwitchFeatures;
 
+import java.util.Optional;
+
 public interface SwitchFeaturesRepository extends Repository<SwitchFeatures> {
+    Optional<SwitchFeatures> findBySwitch(Switch theSwitch);
```
[the remark this answers is not in the document]
Comment (timofei-durakov) [reply services/src/kilda-core/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/SwitchFeaturesRepository.java]: Done

**Pull request merged**